### PR TITLE
Render Fn::Join and Fn::Select

### PR DESCRIPTION
**Pull request by Saarett**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. Render Fn::Join and Fn:Select
2. Tests for Fn::Join and Fn::Select rendering
3. Fixed an issue where rendered attributes (of other intrinsic function) would not be set to the right vertex attributes
4. Fixed the initialization of CloudformationGraphManager in tests (credit goes to @rotemavni for pointing this out)

### CODE REVIEW

File: checkov/common/graph/graph_builder/graph_components/blocks.py [hunk 1]
```diff
@@ -93,54 +95,34 @@ def get_hash(self) -> str:
...
-            elif curr_key in nested_attributes.keys():
-                self.update_inner_attribute(".".join(split_key[i:]), nested_attributes[curr_key], value_to_update)
+    @staticmethod
+    def _should_add_previous_breadcrumbs(change_origin_id: Optional[int],
+            previous_breadcrumbs: List[BreadcrumbMetadata], attribute_at_dest: Optional[str]):
+        return not previous_breadcrumbs or previous_breadcrumbs[-1].vertex_id != change_origin_id
+
+    @staticmethod
+    def _should_set_changed_attributes(change_origin_id: Optional[int], attribute_at_dest: Optional[str]):
+        return True
```
Comment (gruebel) [inline checkov/common/graph/graph_builder/graph_components/blocks.py:125]: is this pre-work for a future update?

**Review by gruebel — COMMENTED** [review]
In general looks good 👍

**Review by nimrodkor — APPROVED** [review]
Very nice! Please tend to @gruebel 's comments

File: checkov/common/graph/graph_builder/graph_components/blocks.py (hunk 1 above)
Comment (Saarett) [reply checkov/common/graph/graph_builder/graph_components/blocks.py:125]: `CloudformationBlocks` class overrides this method but `TerraformBlocks` uses this method as is

File: checkov/cloudformation/graph_builder/variable_rendering/renderer.py [hunk 2]
```diff
@@ -13,18 +13,27 @@
...
+    VERTEX_EVALUATION_CFN_FUNCTIONS = (IntrinsicFunctions.SELECT, IntrinsicFunctions.JOIN)
 
     def __init__(self, local_graph: "CloudformationLocalGraph") -> None:
         super().__init__(local_graph)
-        self.evaluation_methods = {
+        self.edge_evaluation_methods = {
             IntrinsicFunctions.REF: self._evaluate_ref_connection,
             IntrinsicFunctions.FIND_IN_MAP: self._evaluate_findinmap_connection,
             IntrinsicFunctions.GET_ATT: self._evaluate_getatt_connection,
             # ConditionFunctions.IF: self._evaluate_if_connection,
```
[the remark this answers is not in the document]
Comment (Saarett) [reply checkov/cloudformation/graph_builder/variable_rendering/renderer.py:26]: right :)

**Pull request merged**